posthog/hogql/bytecode/operation.py

```diff
@@ -0,0 +1,33 @@
+from enum import Enum
+
+from posthog.hogql.ast import CompareOperationOp, BinaryOperationOp
+
+HOGQL_BYTECODE_IDENTIFIER = "_h"
+
+
+class Operation(str, Enum):
+    PLUS = BinaryOperationOp.Add
+    MINUS = BinaryOperationOp.Sub
+    MULTIPLY = BinaryOperationOp.Mult
+    DIVIDE = BinaryOperationOp.Div
+    MOD = BinaryOperationOp.Mod
+    AND = "and"
+    OR = "or"
```

Comment:
I actually disagree here. I had it at the beginning, but it was just so cumbersome to write AST nodes this explicit. These shorthands `ast.And(exprs=[])` made working with the AST much more approachable. I don't think joining these into one type removes that much complexity 🤷

Reply:
Another thing that annoys me is that we have this duplication:
```py
LIKE = CompareOperationOp.Like
ILIKE = CompareOperationOp.ILike
NOT_LIKE = CompareOperationOp.NotLike
NOT_ILIKE = CompareOperationOp.NotILike
IN = CompareOperationOp.In
NOT_IN = CompareOperationOp.NotIn
REGEX = CompareOperationOp.Regex
NOT_REGEX = CompareOperationOp.NotRegex
```
but then also:
```
NOT = "not"
```

I'm not sure if we should remove the separate `NotFoo` operators, or not. Removing them will make the bytecode larger, and slower, maybe, with the extra step.

The logical conclusion of this train of thought is to also remove the `!=` operator 🤔

Reply:
After sleeping over it, I decided that we're likely better off leaving these as explicit as we can. This will help with performance optimizations later on, as we'll have an easier time targeting specific instructions. E.g. `notEquals` and `notLike` are ClickHouse built in. We weren't using them before, but now we are.